### PR TITLE
Add codecov secret token to CI template

### PR DESCRIPTION
**Pull request by gdalle**

### CODE REVIEW

**Comment by gdalle** [comment]
@DilumAluthge could it be that this fails because you [added an org token](https://github.com/JuliaCI/PkgTemplates.jl/pull/457#issuecomment-1928515545) and so we need to delete the repo token?

File: templates/github/workflows/CI.yml [hunk 1]
```diff
@@ -57,9 +57,11 @@ jobs:
       - uses: julia-actions/julia-runtest@v1
       <<#HAS_CODECOV>>
       - uses: julia-actions/julia-processcoverage@v1
-      - uses: codecov/codecov-action@v3
+      - uses: codecov/codecov-action@v4
         with:
           files: lcov.info
+          token: ${{ secrets.CODECOV_TOKEN }}
+          fail_ci_if_error: true
```
Comment (gdalle) [inline templates/github/workflows/CI.yml:64]: not sure whether this is the best choice, `false` is more resilient to Codecov server failures

Comment (oxinabox) [reply templates/github/workflows/CI.yml:64]: I think better to not fail CI if it errors

**Review by oxinabox — APPROVED** [review]
feel free to merge once working

**Comment by gdalle** [comment]
We didn't get a coverage report on this PR, and the [Codecov CI log](https://github.com/JuliaCI/PkgTemplates.jl/actions/runs/7826346203/job/21352033550?pr=458) says invalid token. Gonna try deleting the repo token to see if using the org-wide solves it

**Comment by gdalle** [comment]
there we go

**Pull request merged**